hibernate-core/src/main/java/org/hibernate/engine/spi/ExecutableList.java

```diff
@@ -298,7 +301,7 @@ public void writeExternal(ObjectOutput oos) throws IOException {
 		else {
 			oos.writeInt( querySpaces.size() );
 			// these are always String, why we treat them as Serializable instead is beyond me...
-			for ( Serializable querySpace : querySpaces ) {
+			for ( Serializable querySpace : NullnessUtil.castNonNull( querySpaces ) ) {
```

Comment:
I don't sprinkle castNonNulls around just because I like 'em so much you know ;)
This is another case where CF doesn't infer the non-nullity correctly, i would even argue that it's a bug, because it only complains about the iteration, NOT about the dereferencing of queryspaces in the line above the iteration ...